### ISSUE TITLE
Problem with `schemaName`

### ISSUE THREAD

**Issue by pchmelar** [apollographql/apollo-ios#2546]
## Bug report
- Thanks for the 1.0.0 release! ❤️  I just integrated it into our testing project, and I have one particular problem with `schemaName`

## Steps to reproduce
- I'm using [`ModuleType.embeddedInTarget`](https://www.apollographql.com/docs/ios/code-generation/codegen-configuration#embedded-in-target) and [`TestMockFileOutput.absolute`](https://www.apollographql.com/docs/ios/code-generation/codegen-configuration#absolute-test-mocks-output) options.
- When `schemaName` is in lowercase (`"schemaName": "rocket"` for example), the `rocket.graphql.swift` with the `Rocket` caseless namespace enum is created. However, the generated Mocks are not working, because there is `rocket` instead of `Rocket`.
- When I change the `schemaName` to start with the capitalized letter (`"schemaName": "Rocket"` for example), the file with caseless namespace enum isn't created at all. 😔 

![New Project](https://user-images.githubusercontent.com/3842139/193872790-99b2f263-9433-4552-8506-1d7367367d3f.png)
<img width="1049" alt="Screenshot 2022-10-04 at 14 21 02" src="https://user-images.githubusercontent.com/3842139/193872686-8625e0e3-fdae-4098-9ac3-556b7ff9192b.png">

## Versions
- `apollo-ios` SDK version: 1.0.0
- Xcode version: 14.0
- Swift version: 5.7
- Package manager: SPM

**Comment by AnthonyMDev** [comment]
Thanks for alerting us to this one. Supports a lowercase named schema here should be an easy fix, we'll get a PR up for that ASAP and release it as part of the first patch version. (We're at GraphQL summit this week, so that will probably be on Monday when we're back!)

> When I change the schemaName to start with the capitalized letter ("schemaName": "Rocket" for example), the file with caseless namespace enum isn't created at all. 😔

This definitely shouldn't be happening. I'm not sure why it would not create the enum. I'll see if we can reproduce that...

**Comment by calvincestari** [comment]
@SilverTab are you able to share with me a project (schema + operations + config) that demonstrates the second issue - missing file? I can replicate the first issue of test mocks using the incorrectly cased type but not the second missing file bug.

**Comment by AnthonyMDev** [comment]
@calvincestari so this wasn't due to having a type named `Rocket` then?

**Comment by SilverTab** [comment]
@calvincestari I'll try to put something together! In the meantime, if that helps; it seems like I didn't have any issues (with lowercase or uppercase schema name) before I started using Fragments in my project 🤔 That's when the error showed up.

**Comment by calvincestari** [comment]
For transparency of where we are on this issue:
* I can replicate the lowercase mock object issue and it's down to inconsistency of where we uppercase the schema name type, we'll get that fixed.
* There is a bug where using a schema name the same name as a type will cause build errors, we'll get that fixed.
* I cannot replicate a missing file though, this is the one that I'm still searching for.

**Comment by pchmelar** [comment]
@calvincestari Yup, I can confirm that. Once I've added `"pruneGeneratedFiles": false` to my JSON configuration the file is not missing. Hope that helps with fixing the issue. 🙂

**Comment by calvincestari** [comment]
> @calvincestari Yup, I can confirm that. Once I've added `"pruneGeneratedFiles": false` to my configuration, the `Rocket.graphql.swift` is successfully generated. Hope that helps with fixing the issue. 🙂

Awesome, that's a solid lead on getting it resolved. Thanks for trying it!

**Comment by SilverTab** [comment]
Same here! `"pruneGeneratedFiles": false` fixes it! :)

**Comment by indama** [comment]
Hi guys, 

Using pod version of Apollo here. 

I have another issue: 

Config:
<img width="691" alt="image" src="https://user-images.githubusercontent.com/4445931/195973239-54b27c60-1f0b-4bb7-b425-3a341fe9e42d.png">

Errors:
<img width="1047" alt="image" src="https://user-images.githubusercontent.com/4445931/195973264-dca672b8-6b03-4eb7-bf7f-325760dd8e4e.png">

<img width="1005" alt="image" src="https://user-images.githubusercontent.com/4445931/195973303-93b63dea-2438-4e96-a773-f72035924d6a.png">

**Comment by AnthonyMDev** [comment]
You need to add `cocoapodsCompatibleImportStatements: true` to your config options.
See the config docs here: https://www.apollographql.com/docs/ios/code-generation/codegen-configuration/#output-options

When you init the codegen config using the CLi built by Cocoapods, it adds this option automatically, so I'm kinda interested to hear how you set this up to not have the option already there!

FWIW, the `SchemaConfiguration` file and any of your CustomScalars won't get overwritten, so you'll need to just change those manually to `import Apollo` instead of `import ApolloAPI`. (We're going to work on fixing that issue as well soon.)

**Comment by indama** [comment]
> You need to add `cocoapodsCompatibleImportStatements: true` to your config options. See the config docs here: https://www.apollographql.com/docs/ios/code-generation/codegen-configuration/#output-options
> 
> When you init the codegen config using the CLi built by Cocoapods, it adds this option automatically, so I'm kinda interested to hear how you set this up to not have the option already there!
> 
> FWIW, the `SchemaConfiguration` file and any of your CustomScalars won't get overwritten, so you'll need to just change those manually to `import Apollo` instead of `import ApolloAPI`. (We're going to work on fixing that issue as well soon.)

Thank you very much. It is working but every time I generate I get import embeddedInTarget.Name. And it is giving error. If I comment out it everything working. 

<img width="768" alt="image" src="https://user-images.githubusercontent.com/4445931/196599902-51b1c7c3-ce07-457d-8a14-c3e97953fd4a.png">

**Comment by calvincestari** [comment]
@indama I think you're using the wrong schema module type. If you're using a CocoaPods setup you need to be using `.other` for the schema module type and not `.embeddedInTarget` - this is detailed in the [documentation](https://www.apollographql.com/docs/ios/code-generation/codegen-configuration#other-schema-module-types).

> This option should be used when you would like to define a schema module using another package management system, **such as CocoaPods**, or manually creating Xcode targets.

**Comment by calvincestari** [comment]
@AnthonyMDev we should probably get a config validation check to ensure you've got the right combination of `.other` when using `cocoapodsCompatibleImportStatements`.

**Comment by AnthonyMDev** [comment]
I don't think that's correct in all cases @calvincestari. We'll have to have a chat about this and make sure we are validating the logic properly.

**Comment by calvincestari** [comment]
1.0.2 included a fix for the first two problems we identified in this issue:
- #2586 
- #2589 

There is one bug left before we can close out this issue and that is the missing file; [some users have indicated](https://github.com/apollographql/apollo-ios/issues/2546#issuecomment-1278024475) that `pruneGeneratedFiles` is causing the bug. We'll do our best to get a fix out for that in `1.0.3`.

**Comment by calvincestari** [comment]
I'm closing this issue since we've addressed the initial issues and we will track the bug related to `pruneGeneratedFiles` in https://github.com/apollographql/apollo-ios/issues/2665 instead.
